.github/workflows/e2e-k8s-main.yaml

```diff
@@ -0,0 +1,145 @@
+#
+# Copyright (C) 2025 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: e2e-k8s-main
+
+on:
+  push:
+    branches:
+      - main
+
+  workflow_dispatch:
+    inputs:
+      organization:
+        default: 'podman-desktop'
+        description: 'Organization of the extension-minikube repository'
+        type: string
+        required: true
+      repositoryName:
+        default: 'extension-minikube'
+        description: 'Podman Desktop Minikube extension repository name'
+        type: string
+        required: true
+      branch:
+        default: 'main'
+        description: 'Podman Desktop Minikube extension repo branch'
+        type: string
+        required: true
+      minikube_driver: 
+        description: 'Minikube cluster driver'
+        type: choice
+        options:
+        - docker
+        - podman 
+        required: true
+
+jobs:
+  e2e-tests:
+    name: Run E2E tests
+    runs-on: ubuntu-24.04
```

Comment:
What version of podman do you need? Here is it the one available in gh ubuntu runner, I think 4.9.3?